services/debugger/cache/cache.go

```diff
@@ -0,0 +1,46 @@
+package cache
+
+//go:generate mockgen -destination=../../../mocks/services/debugger/cache/mock_cache.go -package=mocks_cache github.com/rudderlabs/rudder-server/services/debugger/cache CacheAny
+
+import (
+	"github.com/rudderlabs/rudder-server/services/debugger/cache/embeddedcache"
+	"github.com/rudderlabs/rudder-server/services/debugger/cache/listcache"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+var pkgLogger logger.Logger
+
+type CacheAny interface {
+	Cache[any]
+}
+
+type Cache[E any] interface {
+	Update(key string, value E)
+	ReadAndPopData(key string) []E
+}
+
+type Impl[E any] struct{}
```

Comment:
I think we can remove this `Impl` type and unexport all other types (i.e. `EmbeddedCache`, `ListCache`...) beside `Cache[E]`, right?